### PR TITLE
Finish w_007 and import data

### PR DESCRIPTION
**Pull request by joshlory**
https://decomp.me/scratch/Zn2uO

### CODE REVIEW

File: src/weapon/w_007.c [hunk 1]
```diff
@@ -67,11 +239,100 @@ void EntityWeaponAttack(Entity* self) {
...
+
+            self->animSet = self->ext.weapon.parent->animSet;
+            self->anim = D_35000_8017AB44;
+            self->zPriority = 0xA8;
+            self->unk5A = self->ext.weapon.parent->unk5A;
+            self->palette = self->ext.weapon.parent->palette;
+
+            SetSpeedX(FIX(4.75));
+            temp_v1_4 = self->params & 0xFF00;
+            if ((temp_v1_4 == 0x300) || (temp_v1_4 == 0x600)) {
```
Comment (bismurphy) [inline src/weapon/w_007.c:271]: Looks like temp_v1_4 is not a real variable, I tried removing it in the scratch and still matched.

File: src/weapon/w_007.c [hunk 2]
```diff
@@ -67,11 +239,100 @@ void EntityWeaponAttack(Entity* self) {
...
+            return;
+        }
+        DestroyEntity(self);
+        return;
+    case 1:
+        if (self->hitFlags) {
+            self->step = 3;
+        }
+        DecelerateX(FIX(7) / 32);
+        DecelerateY(FIX(7) / 32);
```
Comment (bismurphy) [inline src/weapon/w_007.c:298]: This should be FIX(7.0 / 32), or FIX(28.0/128). The idea of FIX is to transform a fixed-point value, so we should include the integer and fractional portions together.

File: src/weapon/w_007.c [hunk 3]
```diff
@@ -67,11 +239,100 @@ void EntityWeaponAttack(Entity* self) {
...
+        self->rotY -= 0xC;
+        if (self->rotX < 0x10) {
+            DestroyEntity(self);
+            return;
+        }
+        // fallthrough
+    case 2:
+        self->velocityY = rsin(self->ext.timer.t);
+        self->posY.val += self->velocityY;
+        self->ext.timer.t += 0x20;
```
Comment (bismurphy) [inline src/weapon/w_007.c:321]: Should use `ext.weapon` in weapon functions

**Review by bismurphy — COMMENTED** [review]
Looks good! Some comments on various things, these are just style preferences and if you disagree you can ignore them, but generally I think they would be improvements.

**Review by sozud — APPROVED** [review]

**Pull request merged**